### PR TITLE
Typo correction in variable declaration, and closes feature requests #104, #86, #72 and fix for issue #115, updated excludeMatch() for faster regex check.

### LINKED ISSUE #72: Visual feedback when rule is applied
There needs to be a way for users to know when requests have been manipulated. Since there's no feedback about applied rules, we need to figure out by ourselves when and which of our rules is breaking  something. That can be a burden when you have other add-ons more likely to cause problems.

For example, you could add a small icon over the extension button if a request has affected the current tab.

Ideally, there would also be a logger similar to the one on uBlock Origin for us to see a history of redirections. It works only when you explicitly  open them, so https://github.com/einaregilsson/Redirector/issues/54#issuecomment-215402317 wouldn't be an issue.

### LINKED ISSUE #86: Add option to automatically sync settings between devices
Related: #3

### LINKED ISSUE #104: Request: Comment field?
I could really use a free-form comment field for each redirect.  I've got some complicated regexps that need documentation because I won't be able to easily understand 6 months from now.

Is there any chance of a multi-line textbox for comments getting added to the edit form?

Thanks

### PR DESCRIPTION
**Pull request by gkrishnaks**
1) Typo correction - removed 2nd var on same variable in scope.

2) Closes #104  : User requested for a field to enter description of patterns (or other info about a rule)  https://github.com/einaregilsson/Redirector/issues/104

3) Closes #86 : User requested for Storage Sync while retaining default storage to Local and provide an option to switch to Synced storage.  

4) Fix for issue #115 - Responsive images are now getting redirected in both firefox and chrome.

5) Closes #72 - User requested for visual notification for Successful Redirects. Uses notifications API.

6) Updated excludeMatch() for faster regex check. Replaced exec() with test() as we just need to receive true or false. As Per MDN, regex.test is faster than exec. Should be beneficial for large patterns.

### CODE REVIEW

File: redirector.html [hunk 1]
```diff
@@ -166,6 +174,9 @@ <h4><span class="disabled-marker" ng-show="r.disabled">[Disabled] </span><span>{
 						<div ng-show="r.excludePattern">
 							<label>excluding:</label><p>{{r.excludePattern}}</p>
 						</div>
+						<div ng-if="r.patternDesc">
+							<label>Hint:</label><p>{{r.patternDesc}}</p>
```
Comment (gkrishnaks) [inline redirector.html]: (Labelled it "Hint" here as "Pattern Description" seemed lengthy and didn't fit with other labels)

**Comment by gkrishnaks** [comment]
Also checked this case: 

*Installed add-on  from AMO.
*Setup a redirect rule.
*Built changes locally, updated version in manifest and installed over existing add-on.
*New version retained old redirect rules.
*Old rule continued to work.
*Upon editing an old rule, it added the new field to redirect rule as expected and continued to work.

**Comment by einaregilsson** [comment]
@gkrishnaks I appreciate the interest, will look at it soon as well as the discussion I see is happening on the other issues, I just don't have time right now, just released a game that I have to spend all my time on for the time being :)

**Comment by gkrishnaks** [comment]
@einaregilsson  Committed the Sync feature request also and sent PR!

**Comment by gkrishnaks** [comment]
Sharing fix for issue #115  - Responsive images are now getting redirected in both firefox and chrome.

**Comment by gkrishnaks** [comment]
Closes #72 - User requested for visual notification for Successful Redirects. Uses notifications API.
In Chrome, uses List type notification. Other browsers don't support it, use Basic type notifications instead.

**Comment by gkrishnaks** [comment]
As per Mozilla documentation, regex test() is faster than exec(). 
For ExcludeMatch, we just need to receive true or false and that can be done faster with test() for large exclude patterns.

**Comment by gkrishnaks** [comment]
@ZIEXED Committed a change in css. Increased the height of delete form to allow space for new field Pattern Description.

**Comment by einaregilsson** [comment]
Thanks for all this. Will try to look at it later this week and hopefully get a release out.

**Comment by gkrishnaks** [comment]
Hi @einaregilsson , did you get a chance to look into this? :D

**Comment by einaregilsson** [comment]
Has all this been tested in both Firefox and Chrome?

**Comment by gkrishnaks** [comment]
Yes @einaregilsson , all changes work fine in Firefox and Chrome. 

@ZIEXED  if you would be able to take a look review once, would be great.

**Comment by einaregilsson** [comment]
Ok, finally got around to this, sorry for the delay, and thanks for your contribution. Have merged and published for Chrome, Firefox and Opera, might take an hour or so for it to show up.

**Pull request merged**